### PR TITLE
obs-outputs: Add eRTMP multitrack audio support

### PR DESCRIPTION
**Pull request by palana**
### Description
In addition to multitrack video, eRTMP also includes support for multitrack audio, thus making things like VOD track available in the spec, rather than relying on FLV byte arrays

### Motivation and Context
VOD track currently uses FLV byte arrays to carry the additional audio track, where byte arrays are part of the spec, but using them for audio isn't. With the updated eRTMP multitrack audio support general overhead is lower and also the transport format is part of an open specification.

### How Has This Been Tested?
eRTMP multitrack audio is being tested internally at Twitch

### Types of changes
- New feature (non-breaking change which adds functionality)

### Checklist:

- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [x] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.
- [x] I have included updates to all appropriate documentation.

### CODE REVIEW

**Comment by RytoEX** [comment]
#10500 has been merged, so this can be rebased and undrafted.

**Review by RytoEX — CHANGES_REQUESTED** [review]
Commit message nit:
>  obs-outputs: Add erRTMP multitrack audio support

erRTMP -> eRTMP

File: plugins/obs-outputs/flv-mux.c
Comment (palana) [reply plugins/obs-outputs/flv-mux.c]: @RytoEX, @PatTheMav: added a fixup commit along these lines

you should have commit access to this branch, so feel free to squash/remove as needed

Comment (derrod) [reply plugins/obs-outputs/flv-mux.c]: Having the bitshift here is consistent with how it's done for video and makes more sense than the raw value as it tells you what the spec-defined value is at a glance.

Comment (RytoEX) [reply plugins/obs-outputs/flv-mux.c]: Okay. I'm fine with this as-is, as it's a minor thing that we could change later if needed.

Comment (RytoEX) [reply plugins/obs-outputs/flv-mux.c:43]: Per off-thread discussion, I went ahead and dropped the commit that made the change I had suggested.

**Comment by RytoEX** [comment]
For what it's worth, I tested this and it worked as expected/configured.

**Review by RytoEX — APPROVED** [review]
Tentatively approving this. Tests seemed fine. Would appreciate a second review.

**Comment by RytoEX** [comment]
Holding this due to a confirmed issue regarding audio/video desync that is unique to this implementation.

**Comment by koolscooby** [comment]
Adding the link to the public [alpha Enhanced RTMP v2 spec](https://veovera.org/docs/enhanced/enhanced-rtmp-v2#enhanced-audio) for posterity.

**Comment by palana** [comment]
> Holding this due to a confirmed issue regarding audio/video desync that is unique to this implementation.

@RytoEX: I just pushed a fix for the desync, it was an issue in librtmp's handling of type 3 chunks -> undrafting

**Comment by RytoEX** [comment]
Not sure if the librtmp commit should be first or not. I wouldn't want the eRTMP multitrack audio commit to exhibit the buggy behavior, but I wasn't sure if the librtmp commit has other effects on the existing VOD Audio Track implementation.

**Review by Lain-B — APPROVED** [review]
Great work. Ryan's suggestion to put the commit at the beginning rather than the end might be good but it doesn't really prevent merging of this PR or anything. We'll wait for just a bit more verification/testing to make sure everything's okay but otherwise seems good to go.

**Comment by palana** [comment]
> Not sure if the librtmp commit should be first or not. I wouldn't want the eRTMP multitrack audio commit to exhibit the buggy behavior, but I wasn't sure if the librtmp commit has other effects on the existing VOD Audio Track implementation.

agreed, will move it to the beginning

it doesn't affect the current vod track (as far as I can tell) because the current vod track uses `RTMP_PACKET_TYPE_INFO` [^1]
which always ends up as `RTMP_PACKET_SIZE_LARGE` [^2] and `RTMP_PACKET_SIZE_LARGE` always carries an absolute timestamp and is never being considered for header compression [^3]

[^1]: https://github.com/obsproject/obs-studio/blob/27fa9b1eedbed06753b8d1e090aa760987ff3e46/plugins/obs-outputs/flv-mux.c#L787
[^2]: https://github.com/obsproject/obs-studio/blob/27fa9b1eedbed06753b8d1e090aa760987ff3e46/plugins/obs-outputs/librtmp/rtmp.c#L5344-L5346
[^3]: https://github.com/obsproject/obs-studio/blob/27fa9b1eedbed06753b8d1e090aa760987ff3e46/plugins/obs-outputs/librtmp/rtmp.c#L4118

**Pull request merged**
